_re2.cc

```diff
@@ -813,12 +855,18 @@ regexp_set_add(RegexpSetObject2* self, PyObject* pattern)
     return NULL;
   }
 
+  Py_ssize_t len_pattern;
+#if PY_MAJOR_VERSION >= 3
+  const char* raw_pattern = PyUnicode_AsUTF8AndSize(pattern, &len_pattern);
+#else
   const char* raw_pattern = PyString_AsString(pattern);
+#endif
   if (!raw_pattern) {
     return NULL;
   }
-  Py_ssize_t len_pattern = PyString_GET_SIZE(pattern);
-
+#if PY_MAJOR_VERSION <3
```

Comment:
nit: `< 3`

I think I'd prefer that the `len_pattern` check be moved into the same block, even if it results in the `if (!raw_pattern)` check being duplicated.

Reply:
That's fair. Less macro branching++.